core/state/state_object.go

```diff
@@ -263,9 +262,20 @@ func (self *StateObject) Code(db trie.Database) []byte {
 	return code
 }
 
+// CodeSize returns the size of the contract code associated with this object.
+func (self *StateObject) CodeSize(db trie.Database) int {
+	if self.data.codeSize == nil {
+		self.data.codeSize = new(int)
```

Comment:
Well, using naked constructors everywhere where there are critical fields that have to be set correctly is another antipattern. :)